hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSImageFormatProtobuf.java

```diff
@@ -172,13 +172,55 @@ public LoaderContext getLoaderContext() {
       return ctx;
     }
 
+    /***
+     * a thread for parallel MD5 computing to increase performance when loading
+     */
+    private static class DigestThread extends Thread {
+      volatile private IOException ioe = null;
```

Comment:
Javadocs of the style:
/** Exception thrown when computing the diggest if it cannot be calculated. */
private volatile IOException ioe = null;
/** Calculated digest if there are no error. */
private volatile MD5Hash digest = null;

Note that I think the style says "private volatile"